gem5 fixes and one AndroidDevice fix

Sticklyman1936 opened the pull request:
gem5:
 * Removed busybox dependency for gem5 devices
 * Moved resize shell commands to own method
 * Fixed rare shell connection issues by trying up to 10 times to connect to the simulated system's shell
 * Added the m5 binary to Workload Automation. This is now deployed to the device automatically. The m5 binary is now a partial dependency as it is required for checkpoint support.
 * No longer wait for disabled boot animation
 * Allow the user to specify a login prompt and a corresponding password prompt from the agenda.
 * Removed unused methods
 * Remove the rename in pull_file to align with gem5

Android:
 * Use content instead of settings to get ANDROID_ID

setrofim commented on wlauto/common/gem5/device.py:
```diff
@@ -384,8 +393,11 @@ def push_file(self, source, dest, **kwargs):
-        self.gem5_shell("busybox cp /mnt/obb/{} {}".format(filename, dest))
-        self.gem5_shell("busybox sync")
+        if self.busybox:
+            self.gem5_shell("busybox cp /mnt/obb/{} {}".format(filename, dest))
```
should use `self.busybox` as above.

setrofim commented on wlauto/common/gem5/device.py:
```diff
@@ -618,6 +631,19 @@ def sync_gem5_shell(self):
+        # Try and avoid line wrapping as much as possible. Don't check the error
+        # codes from these command because some of them WILL fail.
+        self.gem5_shell('stty columns 1024', check_exit_code=False)
+        self.gem5_shell('busybox stty columns 1024', check_exit_code=False)
```
`self.busybox`

setrofim commented on wlauto/common/gem5/device.py:
```diff
@@ -618,6 +631,19 @@ def sync_gem5_shell(self):
+        self.gem5_shell('stty columns 1024', check_exit_code=False)
+        self.gem5_shell('busybox stty columns 1024', check_exit_code=False)
+        self.gem5_shell('stty cols 1024', check_exit_code=False)
+        self.gem5_shell('busybox stty cols 1024', check_exit_code=False)
```
`self.busybox`

setrofim commented on wlauto/devices/android/gem5/__init__.py:
```diff
@@ -146,8 +156,11 @@ def install_executable(self, filepath, with_name=None):
-        self.execute('busybox chmod 0777 {}'.format(on_device_executable))
+        if self.busybox:
+            self.execute('busybox cp {} {}'.format(on_device_file, on_device_executable))
+        else:
```
`self.busybox`

Sticklyman1936 commented:
I have addressed the above issues and have added two new patches for you to review. The new patches are:

 * Gem5Device: Removed unused methods	d635596
 * Gem5Device: Remove the rename in pull_file to align with gem5	72a617c
